packages/acceptance-tests/pkg-tests-specs/sources/commands/version/check.test.js

```diff
@@ -123,6 +123,32 @@ describe(`Commands`, () => {
         await run(`version`, `check`);
       }),
     );
+
+    test(`it shouldn't throw if changes were reverted`, makeTemporaryEnv({
```

Comment:
Is there a reason for using `makeTemporaryEnv` instead of `makeVersionCheckEnv` (which does all the initial setup, like `git config` & all)?

Reply:
Personal preference. These setup functions tend to grow with time until any given test uses only a fraction of the actual setup. This makes refactoring later on very hard since it's not obvious anymore what the test case actually tests. But as I said: Capital "IMO". I can reuse the setup if you don't agree.

Reply:
If understand, but the other side of the coin is that it may become hard for people unfamiliar with the test to figure out what's important in the test and what's just setup 🤔

Plus I'd prefer we use the same code style for all the tests in a same file, as much as possible.

Reply:
> what's important in the test and what's just setup

The point is that setup is important for the test (see arrange-act-assert). Since this is not shared every statement is important. For a shared setup you can't be sure if it's relevant or not. 

But I'll use the setup function. I hope nobody thinks the worktree is integral to the test.

Reply:
The big difference in this test is that I can't just simulate a change with the current utils. I do this right now because I can add a file in the initial commit and then simply overwrite it. 

With the test utils I would have to to write to the package json (which means  I have to read first). Could not find an `unlink`. And then I have to be aware of the whole worktree.

It's exactly why these setup functions are an issue because you spend more time pressing the test into the setup mold to the point were any reader has to be aware of the separated setup instead of locating it simply within the test suite. And it's exactly why nobody likes to write tests. Because every test has to use the existing setup in the repository even though it isn't necessary.

I'm not spending more time looking through other tests because that's exactly what any reader has to do later on. Sorry I don't have the time for that.

Reply:
No worry, I understand your position - I hope you understand mine too. I've updated the test to use the helper, as I really want to avoid each test hardcoding `git config`, with potentially subtle differences between each one.